### PR TITLE
cEP-0029: Support TOML as a configuration format

### LINKED ISSUE #179: cEP-0029: Support TOML as a configuration format

### PR DESCRIPTION
**Pull request by PrajwalM2212**
This add cEP for Support TOML as a configuration format project

Closes https://github.com/coala/cEPs/issues/179

### CODE REVIEW

**Comment by PrajwalM2212** [comment]
@RaiVaibhav @Nosferatul @ksdme @margobra8

**Comment by Nosferatul** [comment]
It looks good to me, what will happen if the .coafile have bad syntax/ is not written properly?

**Comment by PrajwalM2212** [comment]
@Nosferatul Is this with respect to `Tool to convert .coafile to TOML config file`?

Suppose the required config is
```text
[Default]
files = *.py, coalib/**/*.py, ./coala, tests/**/*.py, docs/conf.py
max_line_length = 80
```

Case 1:
The user gives 
```text
[Default]
files = *.py, coalib/**/*.py, ./coala, tests/**/*.py, docs/conf.py
max_line_length = 
```
coala will generate 
```python
<Setting object(key='max_line_length', value='', origin='/Users/prajwalm/Documents/coala/coala/.coafile', from_cli=False, to_append=False) at 0x111cd27f0>
```

Case 2:
The user gives 
```text
[Default]
files = *.py, coalib/**/*.py, ./coala, tests/**/*.py, docs/conf.py
max_line_length 
```
coala will generate 
```python
<Setting object(key='files', value='*.py, coalib/**/*.py, ./coala, tests/**/*.py, docs/conf.py\nmax_line_length'
```

when both of them should have errored during the parsing phase as they are syntactical errors.  Instead, coala will error during execution

**Since the tool just converts .coafile to toml config file and the existing parser manages to parse even wrong syntax to something valid, the tool will not break down and will generate a valid toml file.**

As far as the problem of bad syntax/ is not written properly is concerned, I feel that wrong syntax must error during parsing and wrong values/settings must error during execution like how toml and yaml parsers do.
In case 2, there is ambiguity as to whether it is a syntax error or value/setting error as coafile does not define as to what the start and end of a list/array are.
I will add a future work/post-GSoC section in the cEP listing the problems in current coafile syntax and the need to develop improved syntax and parsers.

**Comment by Nosferatul** [comment]
Ok, it's cool, just to keep it in mind, maybe after GSoC we can handle this.
I'm not sure why we need @RJ722  and @li-boxuan approval for this pr.
Also @RaiVaibhav should review it too.

**Comment by PrajwalM2212** [comment]
The codebase automatically requests review from  @RJ722 and @li-boxuan whenever a PR is opened it seems.

File: cEP-0029.md
Comment (RaiVaibhav) [inline cEP-0029.md:163]: No need, we will follow the TOML standard format in most cases because current users of TOML will get confused, we will customize only when it is necessary.

(Leave this section here, don't remove)

**Review by RaiVaibhav — APPROVED** [review]
@Nosferatul have already approved the PR that is good enough to merge it

File: cEP-0029.md
Comment (li-boxuan) [inline cEP-0029.md:16]: A redundant 'that'

File: cEP-0029.md
Comment (li-boxuan) [inline cEP-0029.md]:
```suggestion
in a format that they are probably more familiar with.
```
This sounds better to me

File: cEP-0029.md
Comment (li-boxuan) [inline cEP-0029.md]:
```suggestion
It should convert
```

File: cEP-0029.md
Comment (li-boxuan) [inline cEP-0029.md]: I suggest you avoid using your own username 'prajwalm' here

File: cEP-0029.md
Comment (li-boxuan) [inline cEP-0029.md]:
```suggestion
It should convert
```

File: cEP-0029.md
Comment (li-boxuan) [inline cEP-0029.md:344]: Is this section unfinished or deliberately left empty?

**Review by li-boxuan — COMMENTED** [review]
Just a few minor issues

File: cEP-0029.md
Comment (margobra8) [reply cEP-0029.md]: If this project is carried, should the foundation of it -- i.e. adding some example config files -- be started within this project? I feel that some examples and community-provided configurations sould be present from the start.

File: cEP-0029.md
Comment (mooncos) [inline cEP-0029.md:344]: I feel that this project is related but some example config files and some community-provided examples should be there from the start. Also, don't you think that this should be included within the TOML support itself, not as a separate project?

File: cEP-0029.md
Comment (Naveenaidu) [inline cEP-0029.md]: standardised --> standardized . The same change on line 26 below

File: cEP-0029.md
Comment (PrajwalM2212) [reply cEP-0029.md:344]: @margobra8 I will provide example config files and documentation in TOML from the start to help users write configuration files in TOML like http://docs.coala.io/en/latest/Users/coafile.html.

`A repository of good config files to allow users to easily bootstrap their projects.` is about creating a separate repo similar to https://github.com/github/gitignore and https://github.com/moremoban/pypi-mobans to allow easy bootstrapping of projects.

File: cEP-0029.md
Comment (PrajwalM2212) [reply cEP-0029.md]: US vs UK format

File: cEP-0029.md
Comment (mooncos) [reply cEP-0029.md:344]: @PrajwalM2212 the thing I'm trying to get to is if you actually consider generating another repo with templates as part of the initial requirements of the project or as future development or to be implemented later.

Comment (PrajwalM2212) [reply cEP-0029.md:344]: As part of GSoC, another repo will be created with few templates as part of the initial requirements of the project.  A minimal version of `A repository of good config files` will be created during GSoC.

**Review by abhishalya — APPROVED** [review]

**Comment by abhishalya** [comment]
This is now merged as a part of https://github.com/coala/cEPs/pull/197 :)

**Pull request closed without merging**
